pkg/template/admission/impersonation/admission.go

```diff
@@ -0,0 +1,68 @@
+package impersonation
+
+import (
+	"errors"
+	"io"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	"github.com/openshift/origin/pkg/client"
+	templateapi "github.com/openshift/origin/pkg/template/api"
+	userapi "github.com/openshift/origin/pkg/user/api"
+	"k8s.io/kubernetes/pkg/admission"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+)
+
+func init() {
+	admission.RegisterPlugin("openshift.io/TemplateImpersonationAdmissionPlugin", func(c clientset.Interface, config io.Reader) (admission.Interface, error) {
+		return &impersonationAdmissionHandler{
+			Handler: admission.NewHandler(admission.Create, admission.Update),
+		}, nil
+	})
+}
+
+type impersonationAdmissionHandler struct {
+	*admission.Handler
+	oc client.Interface
+}
+
+func (iah *impersonationAdmissionHandler) Admit(attr admission.Attributes) (err error) {
```

Comment:
I want these sorts of things in the strategy, I do not agree that admission control is appropriate for this.  Someone can expose the storage in a different place and miss the admission controller.